### ISSUE TITLE
Harvest Data 2015 - LicAvail Missing

### ISSUE THREAD

**Issue by Huh** [Huh/PopR_SDGFP#123]
I do not see any entries for the LicAvail column in the data.  This is causing the IPM tab to have trouble because this column is needed to calculate success.

Thanks,

Josh

**Comment by HeatherMcW** [comment]
Are you referring to ResTagAvail and NonResTagAvail in the dbo_License and dbo_Tag tables?

**Comment by HeatherMcW** [comment]
I did some spot checking and will fill in the missing records that have number values, but a lot of the entries that are blank correspond to an unlimited number of licenses. How do you want unlimited values to be handled?

**Comment by Huh** [comment]
The site only needs values for "regulated" harvest, so cases where the number of licenses are limited by managers.  The 2014 data is a good example of what is needed.  The unregulated harvest are all 0 and the regulated have values used to determine success.  
Thanks for getting on this so fast Heather!  I may also be bugging you in a separate issue about survival/telemetry records. Josh Nowak 

    On Friday, June 10, 2016 11:03 AM, HeatherMcW <notifications@github.com> wrote:
 

 I did some spot checking and will fill in the missing records that have number values, but a lot of the entries that are blank correspond to an unlimited number of licenses. How do you want unlimited values to be handled?—

**Comment by HeatherMcW** [comment]
so you want be to put 0's in for the unregulated seasons? The 0's that are currently in there are because non-residents aren't allowed to purchase certain licenses.

Side note: the heading says tags available but it really means licenses available. Is this a problem on your end?

**Comment by Huh** [comment]
The tag v license bit is switched in the code on my end, no problem at all.

In the code, anything in this column that is not a number is replaced with 0 for the sake of presentation/calculation, so no need to insert 0.  This is the problem in a nutshell, not that the values are replaced, but that we are trying to divide by 0.  We need values where they exist so I can calculate success.

It looks to me that when I query "GetTags" table the ResTagAvail column contains all blanks except two entries.  The sum of ResTagAvail and NonResTagAvail columns is used as the total number of licenses available.  Success is then calculated as Harvest/TagAvail.  Does that make sense?

Here are the entries we are considering regulated harvest, if that helps...

```r
      regulated <- c("West River Firearms", 
              "East River Firearms", 
              "Refuge", 
              "Black Hills Firearms",
              "Custer State Park", 
              "Special Buck East River",
              "Special Buck West River")
```

**Comment by Huh** [comment]
Oops, the NonResTagAvail column is also all blanks except the last two entries, which are both 0.

**Comment by HeatherMcW** [comment]
I gotcha. I will sure all of those seasons have values before I leave today.

**Comment by HeatherMcW** [comment]
Are you using both the dbo_License and the dbo_Tag tables?

**Comment by Huh** [comment]
Yes, I am using both of them.  I believe there is unique information that requires both.
I get harvest information from the License table and effort/sales information from the Tag table.  Sound right? Josh Nowak 

    On Friday, June 10, 2016 1:53 PM, HeatherMcW <notifications@github.com> wrote:
 

 Are you using both the dbo_License and the dbo_Tag tables?—

**Comment by HeatherMcW** [comment]
no. Deer harvested should all come out of the harvest table. License & Tag tables are very similar to each other. The license table has everything summarized at the unit level (Example: Unit 01A in 2005 has 3 tags 01, 03, 09), but all of the information is summarized as a single entry of unit 01A). and the Tag table has the unit/license information broken out by tag when its available (Unit 01A in the previous example would have 3 separate entries for each tag type). Its kinda confusing because a lot of the time I don't have licensing/harvest information at the tag level but I do have it at the license level.

I didn't realize so many values were missing. I will get what I can done today in the Tag table, but I probably won't be able to finish until Monday or Tuesday.

**Comment by Huh** [comment]
I am sorry, you are right, I use the GetHarvest and GetTags entries to the webservice, not licenses and tags.  Yes, license and tags are similar and tag is most useful to me because of the differences you explained.

Thanks for all of your work, much appreciated!

**Comment by HeatherMcW** [comment]
Available licenses are now entered for all of the regulated seasons in the Tag table. Will you need this information for the License table?

**Comment by Huh** [comment]
No thanks.  Appreciate your work today, thank you! Josh 

    On Friday, June 10, 2016 2:59 PM, HeatherMcW <notifications@github.com> wrote:
 

 Available licenses are now entered for all of the regulated seasons in the Tag table. Will you need this information for the License table?—

**Comment by gfrc11752** [comment]
Shouldn't we be using the tags sold column for the analysis described above? Just because a tag is available doesn't mean it was actually allocated thus having a 0% probability of being filled.

**Comment by Huh** [comment]
Depends, as I understood the original argument, SDGFP can control the number of tags that are available, but not the number sold nor the number of animals harvested.

Perhaps we should revisit the document that describes how I calculated harvest and the like?
